TYPO3.Flow/Classes/TYPO3/Flow/Property/TypeConverter/DateTimeConverter.php

```diff
@@ -138,12 +139,14 @@ public function convertFrom($source, $targetType, array $convertedChildPropertie
             }
             if (isset($source['dateFormat']) && strlen($source['dateFormat']) > 0) {
                 $dateFormat = $source['dateFormat'];
+                $isFormatSpecified = true;
             }
         }
         if ($dateAsString === '') {
             return null;
         }
-        if (ctype_digit($dateAsString) && $configuration === null && (!is_array($source) || !isset($source['dateFormat']))) {
+        $isFormatConfigured = ($configuration !== null && $configuration->getConfigurationValue(DateTimeConverter::class, self::CONFIGURATION_DATE_FORMAT) !== null);
+        if (!$isFormatConfigured && !$isFormatSpecified && ctype_digit($dateAsString)) {
```

Comment:
Well, there is one slim change on top of the "configuration but no format": Previously it also wouldn't trigger if `$source['dateFormat']` was set but empty, which it now will - hence the test failing. But as stated I think this is rather odd behaviour.

I could be convinced to move that additional (breaking?) change to next major only.